joomla/joomla-cms
# [4.0] Remove file from template url

Currently in the template file manager, the URL argument `file` is always present, even when no file is actually edited. This isn't necessary and is a bit confusing in the code.

### Summary of Changes
This PR removes all instances where the argument isn't necessary and adjusts the checks.

There will be a small difference in behavior: Currently you can execute non-edit-related actions (eg create override or delete a folder) and after the redirect the previously opened file will still be open. After this patch it will be closed.
Keep in mind that even currently, all changes made in that file since the last save would be lost if you run such an action. So it gives a wrong impression to always reopen it.
Imho, the edit mode should disable all other tabs and non-edit buttons. To prevent the data loss. But that would be a task for another PR.

I also removed the hidden `compressed_formats` parameter and related checks. Since it was a hidden parameter with only `zip` as value, it basically was a hardcoded to 'zip'. I instead just really hardcoded it into the code.

### Testing Instructions
Play around with the template file manager and see if something is broken.
Especially also try the various file manipulations (edit, create, delete, copy, resize, crop, ...)


### Expected result
Works as before


### Actual result
No difference


### Documentation Changes Required
None
